src/btree/bt_debug.c

```diff
@@ -126,9 +133,10 @@ __debug_item_key(WT_DBG *ds, const char *tag, const void *data_arg, size_t size)
     WT_SESSION_IMPL *session;
 
     session = ds->session;
-
     return (ds->f(ds, "\t%s%s{%s}\n", tag == NULL ? "" : tag, tag == NULL ? "" : " ",
-      __wt_key_string(session, data_arg, size, ds->key_format, ds->t1)));
+      F_ISSET(ds, WT_DEBUG_DUMP_APP_DATA) ?
+        __wt_key_string(session, data_arg, size, ds->key_format, ds->t1) :
+        "redacted"));
```

Comment:
Minor suggestion: The printlog code in `log_auto.c` uses `REDACTED` to make it stand out. You could make it consistent here by capitalizing the word.